tests/end-to-end/test_database.py

```diff
@@ -125,6 +227,103 @@ def test_database_dump_and_restore():
         shutil.rmtree(temporary_directory)
 
 
+def test_database_dump_and_restore_with_restore_cli_arguments():
+    # Create a Borg repository.
+    temporary_directory = tempfile.mkdtemp()
+    repository_path = os.path.join(temporary_directory, 'test.borg')
+    borgmatic_source_directory = os.path.join(temporary_directory, '.borgmatic')
+
+    original_working_directory = os.getcwd()
+
+    try:
+        config_path = os.path.join(temporary_directory, 'test.yaml')
+        write_simple_custom_restore_configuration(
+            temporary_directory, config_path, repository_path, borgmatic_source_directory
+        )
+
+        subprocess.check_call(
+            ['borgmatic', '-v', '2', '--config', config_path, 'init', '--encryption', 'repokey']
+        )
+
+        # Run borgmatic to generate a backup archive including a database dump.
+        subprocess.check_call(['borgmatic', 'create', '--config', config_path, '-v', '2'])
+
+        # Get the created archive name.
+        output = subprocess.check_output(
+            ['borgmatic', '--config', config_path, 'list', '--json']
+        ).decode(sys.stdout.encoding)
+        parsed_output = json.loads(output)
+
+        assert len(parsed_output) == 1
+        assert len(parsed_output[0]['archives']) == 1
+        archive_name = parsed_output[0]['archives'][0]['archive']
+
+        # Restore the database from the archive.
+        subprocess.check_call(
+            [
+                'borgmatic',
+                '-v',
+                '2',
+                '--config',
+                config_path,
+                'restore',
+                '--archive',
+                archive_name,
+                '--hostname',
+                'postgresql2',
+                '--port',
+                '5433',
+                '--username',
+                'postgres2',
+                '--password',
+                'test2',
+            ]
+        )
```

Comment:
With the original database tests, there's only a single database instance per hook, so if the restore executes without erroring, you're pretty well assured that it worked. However with these new tests, there are two database instances per hook. So a silent failure mode is if a restore happens successfully—just to the wrong database instance. Like if all the `restore_*` options or new CLI flags get ignored.

One way to check for that would just be to run a query (`psql` or similar) against the expected database instance and make sure the named database in question actually exists there. If it does, then you know it must've been because you restored it. (But that will only work if the `POSTGRES_DATABASE` in the `docker-compose.yaml` is different for each database instance.)

An argument against doing this is that you've already got coverage for the "wrong database instance" in your unit tests, so maybe you don't need it / want it in your end-to-end tests too. Ultimately, I'll leave the decision to you. Just raising the question here in case you want to do anything about it.

Reply:
Hm, thanks for letting me know. 

Also, won't docker create the POSTGRES_DB in docker_compose.yaml as soon as I start the tests (even before backup)? If yes, even the database check query won't work and I'll have to check for something else inside the database.

Reply:
Yeah, that's why I was suggesting two different `POSTGRES_DB` names, one for each database instance. Like if the first instance's database is `test1` and the second instance's is `test2`. Then if you attempt to restore `test1` to the second instance, and you see `test1` actually created there, you know it must've worked.

Reply:
Doesn't the restore command already need the database to exist?

Reply:
https://www.postgresql.org/docs/current/app-pgrestore.html I think that's what the --create option is for.

Reply:
You are correct! Without that flag, it won't create it. You could _add_ the flag though, which might not be a bad idea to have anyway.. Assuming it still works when the database already exists.

Reply:
Just following up on this. It turns out that restores are [not working correctly in this end-to-end test](https://projects.torsion.org/borgmatic-collective/borgmatic/issues/738#issuecomment-6912) in main—likely broken by me _after_ you submitted this PR. But my point in bringing this up now is that more assertions on whether the tests were actually restoring the expected data probably would've caught that!